apps/backend/app/auth/nextauth_auth.py

```
@@ -0,0 +1,104 @@
+) -> Optional[str]:
+    """
+    Optional authentication - returns user ID if authenticated, None otherwise
+    """
+    if not authorization or not authorization.startswith("Bearer "):
+        return None
+    
+    try:
+        token = authorization.split(" ")[1]
+        credentials = HTTPAuthorizationCredentials(scheme="Bearer", credentials=token)
+        return get_current_user_id(credentials)
+    except:
```

Ruff (0.14.4)

Do not use bare `except`

(E722)